### PR TITLE
chore: libwaku - exposing discv5 store-client and relay connected and mesh peers

### PR DESCRIPTION
**Pull request by Ivansete-status**
# Description
Step further to make nwaku completely accessible from libwaku. Still pending to expose some features such as peer exchange, or filter.

# Changes

- Allow to start or stop discv5
- Expose lightpush request operation
- Expose list of connected and mesh peers
- Expose store client

### CODE REVIEW

**Review by gabrielmer — APPROVED** [review]
LGTM, thanks so much!

File: library/libwaku.nim [hunk 1]
```diff
@@ -383,6 +384,119 @@ proc waku_relay_unsubscribe(
...
+    ctx: ptr Context,
+    pubSubTopic: cstring,
+    jsonWakuMessage: cstring,
+    callback: WakuCallBack,
+    userData: pointer,
+): cint {.dynlib, exportc, cdecl.} =
+  ctx[].userData = userData
+
+  if isNil(callback):
+    return RET_MISSING_CALLBACK
```
Comment (SionoiS) [inline library/libwaku.nim:455]: Why is this needed here but not in all other procs?

**Review by SionoiS — APPROVED** [review]
LGTM Thanks!

File: library/libwaku.nim (hunk 1 above)
Comment (Ivansete-status) [reply library/libwaku.nim:455]: > Why is this needed here but not in all other procs?

Good point! Because some checks are left. I will submit an enhancement in a separate PR
